sycl/doc/SpecializationConstants.md

```diff
@@ -200,93 +200,56 @@ and the user says
+of its return type plus one `__spirv_SpecConstantComposite` to gather members
+back into a single composite. If any composite member is another composite, then
+it will be also represented by number of `__spirv_SpecConstant` plus one
+`__spirv_SpecConstantComposite`:
 
 ```
-%gold_POD_a0x = call i32 __spirv_SpecConstant(i32 10, i32 0)
-%gold_POD_a0y = call float __spirv_SpecConstant(i32 11, float 0)
-%gold_POD_a1x = call i32 __spirv_SpecConstant(i32 12, i32 0)
-%gold_POD_a1y = call float __spirv_SpecConstant(i32 13, float 0)
+%gold_POD_A0_x = call i32 __spirv_SpecConstant(i32 10, i32 0)
+%gold_POD_A0_y = call float __spirv_SpecConstant(i32 11, float 0)
+
+%gold_POD_A0 = call %struct.A __spirv_SpecConstantComposite(i32 %gold_POD_A0_x, float %gold_POD_A0_y)
+
+%gold_POD_A1_x = call i32 __spirv_SpecConstant(i32 12, i32 0)
+%gold_POD_A1_y = call float __spirv_SpecConstant(i32 13, float 0)
+
+%gold_POD_A1 = call %struct.A __spirv_SpecConstantComposite(i32 %gold_POD_A1_x, float %gold_POD_A1_y)
+
+%gold_POD_A = call [2 x %struct.A] __spirv_SpecConstantComposite(%struct.A %gold_POD_A0, %struct.A %gold_POD_A1)
 %gold_POD_b = call i32 __spirv_SpecConstant(i32 14, i32 0)
-```
 
-And 1 "composite"
+%gold = call %struct.POD __spirv_SpecConstantComposite([2 x %struct.A] %gold_POD_A, i32 %gold_POD_b)
 
-```
-  %gold_POD = call %struct.POD __spirvCompositeSpecConstant<POD type mangling>(i32 10, i32 11, i32 12, i32 13, i32 14)
 ```
 
-where `__spirvCompositeSpecConstant<POD type mangling>` is a new SPIR-V intrinsic which
- represents creation of a composite specialization constant. Its arguments are spec
- constant IDs corresponding to the leaf fields of the POD type of the constant.
-Spec ID for the composite spec constant is not needed, as runtime will never use it - it will use IDs of the leaves instead.
- Yet, the SPIR-V specification does not allow `SpecID` decoration for composite spec constants.
+Spec ID for the composite spec constant is not needed, as runtime will never use
+it - it will use IDs of the leaves instead.
+Yet, the SPIR-V specification does not allow `SpecID` decoration for composite
+spec constants, because its defined by its members instead.
 
 ##### The post-link tool changes
 
 For composite specialization constants the post link tool will additionally
-generate linearized list of \<leaf spec ID,type,offset,size\> tuples (descriptors),
+generate linearized list of \<leaf spec ID,offset,size\> tuples (descriptors),
 where each tuple describes a leaf field, and store it together with the
 existing meta-information associated with the specialization constants and
 passed to the runtime. Also, for a composite specialization constant there is
 no ID map entry within the meta information, and the composite constant is
 referenced by its symbolic ID. For example:
 
 ```
-MyConst_mangled [10,int,0,4],[11,float,4,4],[12,int,8,4],[13,float,12,4],[14,int,16,4]
-```
-
-#### LLVMIR-\>SPIR-V translator
```

Comment:
Why is this section removed? Are you proposing not to change LLVM translator at all?

Reply:
> Why is this section removed?

From my point of view it is not really useful, as interface between LLVM IR and SPIR-V is owned by the translator and not by intel/llvm. We use this interface (SPIR-V friendly IR) for almost every SYCL built-in and we don't have an extensive documentation for the format for each built-in, because we just refer to a document in the upstream translator repo.

> Are you proposing not to change LLVM translator at all?

It still needs to be changed, but I don't see anything special which needs to be documented here: I would rather update [SPIR-V friendly IR document](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/docs/SPIRVRepresentationInLLVM.rst) to clarify the exact format if that is needed.

Reply:
I do think that the design doc would benefit a lot from at least high-level description of what happens in the translator, as this is part of compilation process, even though separated into a different component.

Reply:
> I do think that the design doc would benefit a lot from at least high-level description of what happens in the translator, as this is part of compilation process, even though separated into a different component.

Ok, makes sense. I will leave some high-level description of the interface with the SPIR-V translator then

Reply:
Returned this section back in e40b404e10c2bad9b4b40bf501ff6cd2bb17b2f4